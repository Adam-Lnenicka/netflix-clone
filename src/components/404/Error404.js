import React from "react";
import { Link } from "react-router-dom";
import { ButtonSecondary } from "../../styles/global/Buttons";
import { Error404Container, Sign404 } from "./styles/Error404styles";

const Error404 = () => {
  return (
    <Error404Container>
      <p>Page Not Found</p>
      <Sign404>404</Sign404>

      <Link to="/">
        <ButtonSecondary>back to home</ButtonSecondary>
      </Link>
    </Error404Container>
  );
};

export default Error404;
