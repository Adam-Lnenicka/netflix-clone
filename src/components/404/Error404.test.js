import "jsdom-global/register";
import { BrowserRouter as Router } from "react-router-dom";
import toJson from "enzyme-to-json";
import React from "react";
import { Link } from "react-router-dom";
import { mount } from "enzyme";
import Error404 from "./Error404";

describe("404 Page", () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = {
      to: "/",
    };
    wrapper = mount(
      <Router location="home">
        <Error404>
          <Link {...props} />
        </Error404>
      </Router>
    );
  });

  it("Create snapshot of 404 component with link", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
