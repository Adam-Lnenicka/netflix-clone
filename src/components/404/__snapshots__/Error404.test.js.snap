// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`404 Page Create snapshot of 404 component with link 1`] = `
<BrowserRouter
  location="home"
>
  <Router
    location={
      Object {
        "hash": "",
        "key": "default",
        "pathname": "blank",
        "search": "",
        "state": null,
      }
    }
    navigationType="POP"
    navigator={
      Object {
        "action": "POP",
        "back": [Function],
        "block": [Function],
        "createHref": [Function],
        "forward": [Function],
        "go": [Function],
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "default",
          "pathname": "blank",
          "search": "",
          "state": null,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Error404>
      <div
        className="error404"
      >
        <p>
          Page Not Found
        </p>
        <p
          className="sign404"
        >
          404
        </p>
        <Link
          to="/"
        >
          <a
            href="/"
            onClick={[Function]}
          >
            <button
              className="button-secondary"
            >
              back to home
            </button>
          </a>
        </Link>
      </div>
    </Error404>
  </Router>
</BrowserRouter>
`;
